osf/models/preprint_service.py

```diff
@@ -228,9 +228,10 @@ def set_preprint_license(self, license_detail, auth, save=False):
         if save:
             self.save()
 
-    def set_identifier_values(self, doi, ark, save=False):
+    def set_identifier_values(self, doi, ark=None, save=False):
```

Comment:
This is more a question, does `ark` even still need to be in the signature for `set_identifier_values`? How often outside of tests will we ever be setting the values?

Reply:
You should also update the useage of `set_identifier_values` in the `get_and_set_preprint_identifiers` method of `website/preprints/tasks.py` --

        preprint.set_identifier_values(doi=id_dict['doi'], ark=id_dict['ark'], save=True)

 if `parse_identifiers` returns an `id_dict` here with no `ark` key, this would raise a key error